### ISSUE TITLE
All Tags - Contant Fiscal Crisis (Yearly Deficit issue)

### ISSUE THREAD

**Issue by Marok132** [TNODevs/TNO-Bug-Tracker#3872]
### Operating System

Windows 10

### HOI4 version

1.15.2

### TNO version

1.7.0

### Do you have other mods in use?

TNO Tool Mod

### Were you using Steam?

No

### Which major expansions do you NOT have?

I have all

### Describe the bug

After the update was out, I found many AI countries started suffering from constant fiscal crisis. I think the issue is Yearly Deficit because it's huge (Check my screenshots below). 

### Steps to reproduce / Cause of the issue

Idk.

### Expected behaviour

The Yearly Deficit should be SO huge.

### Screenshots

Here's examples of this problem.

![image](https://github.com/user-attachments/assets/a13a5b32-155a-4818-ab6f-b98f5c496b93)
![image](https://github.com/user-attachments/assets/5cf06982-d5c7-484e-8970-60bee3ef7a7f)
![image](https://github.com/user-attachments/assets/ec8006ad-bace-4352-a207-5c7fafd698f2)
![image](https://github.com/user-attachments/assets/370ca2d3-4e29-4ffb-bb4a-05077456f203)
![image](https://github.com/user-attachments/assets/4ba4c7d6-df53-4dbe-8e3d-30ed11ee9ae5)
![image](https://github.com/user-attachments/assets/406c61c2-37d9-4af1-8e21-c1d0497716c6)

**Comment by Marok132** [comment]
> The Britain one was largely caused by their content giving too much debt in too little time, which combined with the Civil War screwed their economy. I did my best to fix it and lower their debt.
> 
> A Fiscal Crisis is not guaranteed just from spending. In fact, the AI is coded to spend a lot on social (and army spending for despotist tags), and once debt high enough, cut down a lot.
> 
> For some countries, there is simply not much they can do to prevent a fiscal crisis due to their army size primarily, but the AI does its best to prevent it.
> 
> A coming update will bring more tools for the AI to use, such as Austerity Measures, War Taxes, other Bank Policies and Tax Hikes/Tax Cuts. From my experience testing those features, they greatly improve the AI's capabilities.

Thanks for the answer!

**Comment by Marok132** [comment]
> The Britain one was largely caused by their content giving too much debt in too little time, which combined with the Civil War screwed their economy. I did my best to fix it and lower their debt.
> 
> A Fiscal Crisis is not guaranteed just from spending. In fact, the AI is coded to spend a lot on social (and army spending for despotist tags), and once debt high enough, cut down a lot.
> 
> For some countries, there is simply not much they can do to prevent a fiscal crisis due to their army size primarily, but the AI does its best to prevent it.
> 
> A coming update will bring more tools for the AI to use, such as Austerity Measures, War Taxes, other Bank Policies and Tax Hikes/Tax Cuts. From my experience testing those features, they greatly improve the AI's capabilities.

![Image](https://github.com/user-attachments/assets/c2c50f2f-d816-4904-9aa0-d447079a3db4)

This problem has not gone yet sadly.

**Comment by Jack121Q** [comment]
Oh the Britain is much better now. They don't have 3 Fiscal Crisises every game and I lowered the debt costs of their decisions.
The AI Tax Measures have been pushed and those should help as well.

**Comment by Marok132** [comment]
> Oh the Britain is much better now. They don't have 3 Fiscal Crisises every game and I lowered the debt costs of their decisions. The AI Tax Measures have been pushed and those should help as well.

I also have the same issues with Romania and Finland btw. Their economy sucks.

**Comment by Marok132** [comment]
> Oh the Britain is much better now. They don't have 3 Fiscal Crisises every game and I lowered the debt costs of their decisions. The AI Tax Measures have been pushed and those should help as well.

Also some countries with low GDP still suffer from constant fiscal as well,

**Comment by Jack121Q** [comment]
It's not a fool-proof system but with the AI Tax Cuts and Hikes it should be better

**Comment by Marok132** [comment]
> It's not a fool-proof system but with the AI Tax Cuts and Hikes it should be better

Will it be coming in the future hotfix? Their economy won't suffer from fiscal, will it?

**Comment by Jack121Q** [comment]
It's coming next patch.

**Comment by Marok132** [comment]
> It's coming next patch.

I hope next patch will be about Brazil. Anyway, TNO still needs a new hotfix because there are too many bugs in the current one. I think you could notice it by checking my issues.

**Comment by Marok132** [comment]
> I can't and won't promise no fiscal crisis, but it should be better.

Anyway thanks for trying to fix them.

**Comment by Jack121Q** [comment]
Sorry you don't like the patch, I tried my best to add as much as I could and fix stuff but sometimes things slip by when I code too much in a day.

**Comment by Jack121Q** [comment]
I do my best to fix them as soon as possible once I learn there's an issue though.

**Comment by Marok132** [comment]
> Sorry you don't like the patch, I tried my best to add as much as I could and fix stuff but sometimes things slip by when I code too much in a day.

Don't apologize, please. I can see you really try to fix all the issues in Github and especially mine, and I'm thankful and I respect you (you're one of few tno devs I really respect, to be honest). Were you the only one who was working on the hotfix? If so, then I'm not surprised because it's huge work for one person.

Anyway, thanks for your work. I appreciate it.

**Comment by Marok132** [comment]
> I do my best to fix them as soon as possible once I learn there's an issue though.

I believe in it. I wish I could hug you as a thank you for fixing my issues.

**Comment by Jack121Q** [comment]
> > Sorry you don't like the patch, I tried my best to add as much as I could and fix stuff but sometimes things slip by when I code too much in a day.
> 
> Don't apologize, please. I can see you really try to fix all the issues in Github and especially mine, and I'm thankful and I respect you (you're one of few tno devs I really respect, to be honest). Were you the only one who was working on the hotfix? If so, then I'm not surprised there are many bugs since it's huge work for one person to fix them.
> 
> Anyway, thanks for your work. I appreciate it.

I didn't work on it alone, but yes, I contributed a lot.

**Comment by Jack121Q** [comment]
> > I do my best to fix them as soon as possible once I learn there's an issue though.
> 
> I believe in it. I wish I could hug you as a thank you for fixing my issues.

I'll likely get to bringing the number down to 150 by the end of the week. I am almost done with a huge pet project you and the community might like.

**Comment by Marok132** [comment]
> > > I do my best to fix them as soon as possible once I learn there's an issue though.
> > 
> > 
> > I believe in it. I wish I could hug you as a thank you for fixing my issues.
> 
> I'll likely get to bringing the number down to 150 by the end of the week. I am almost done with a huge pet project you and the community might like.

Is it Iberia, I guess?

**Comment by Jack121Q** [comment]
No, it's another pet project like the Consequences of War and Capitulation from this patch.

**Comment by Marok132** [comment]
> No, it's another pet project like the Consequences of War and Capitulation from this patch.

Ah, alright. Btw, can we talk in Discord, not here? I do not want to flood here.

**Comment by Jack121Q** [comment]
Sure, but know I can't and won't leak anything. My discord is: kingdragon.

**Comment by Marok132** [comment]
> Sure, but know I can't and won't leak anything. My discord is: kingdragon.
I won't ask you about leaks, don't worry.

**Comment by Marok132** [comment]
> Sure, but know I can't and won't leak anything. My discord is: kingdragon.

Discord says I can't send an invite.

![Image](https://github.com/user-attachments/assets/0c2fdbdf-4ebb-43e1-9df0-352d3ce28695)

**Comment by Marok132** [comment]
> > Sure, but know I can't and won't leak anything. My discord is: kingdragon.
> 
> Discord says I can't send an invite.
> 
> ![Image](https://github.com/user-attachments/assets/0c2fdbdf-4ebb-43e1-9df0-352d3ce28695)

What should I do?

**Comment by Jack121Q** [comment]
Odd. Ping me in a channel on the community server if you can and I'll add you instead.

**Comment by Marok132** [comment]
> Odd. Ping me in a channel on the community server if you can and I'll add you instead.

Did you forget I got banned there? I can just give my discord here; avata_t
